### PR TITLE
Parameterizing orka-image name

### PR DESCRIPTION
**Pull request by hokadiri**
# Changes

**What:**
Parameterizing orka task images so it can be overriden

**Why:**
docker.io not accessible from internal networks. one needs to either access it via a proxy OR copy the image into an internal docker registry. Doing this requires modifying the `image` var.  

**How:**
Creating params var for task image which defaults to orka docker.io image.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Follows the [authoring recommendations](https://github.com/tektoncd/catalog/blob/main/recommendations.md)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [x] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [x] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [x] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [x] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [x] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

File: task/orka-full/0.1/orka-full.yaml [hunk 1]
```diff
@@ -97,14 +102,19 @@ spec:
       type: string
       default: /tekton/home
       description: Absolute path to the user's home directory.
+    - name: task-image
+      type: string
+      description: |
+        The name of the docker image which runs the task step.
+      default: docker.io/macstadium/orka-tekton-runner:2020-10-23-a93757dc-0.1@sha256:e8ed3370dcb91cdb8bcd4e9a7e9f6652879d80acdab9644cda69a98cd8c93339
```
Comment (vinamra28) [inline task/orka-full/0.1/orka-full.yaml:109]: are we using this parameter anywhere?

**Comment by hokadiri** [comment]
Closing #939  in favor of this pr.

File: task/orka-init/0.1/orka-init.yaml [hunk 2]
```diff
@@ -71,9 +76,14 @@ spec:
       description: |
         The name of the key in the VM name config map, which stores the name of the generated VM configuration.
       default: vm-name
+    - name: task-image
+      type: string
+      description: |
+        The name of the docker image which runs the task step.
+      default: docker.io/macstadium/orka-tekton-runner:2020-10-23-a93757dc-0.1@sha256:e8ed3370dcb91cdb8bcd4e9a7e9f6652879d80acdab9644cda69a98cd8c93339
```
Comment (vinamra28) [inline task/orka-init/0.1/orka-init.yaml:83]: same

**Review by vinamra28 — CHANGES_REQUESTED** [review]
can you please address these changes?
/ok-to-test

File: task/orka-deploy/0.1/orka-deploy.yaml [hunk 3]
```diff
@@ -94,14 +99,19 @@ spec:
       type: string
       default: /tekton/home
       description: Absolute path to the user's home directory.
+    - name: task-image
```
Comment (piyush-garg) [inline task/orka-deploy/0.1/orka-deploy.yaml]: This should be removed

File: task/orka-full/0.1/orka-full.yaml [hunk 4]
```diff
@@ -97,14 +102,19 @@ spec:
       type: string
       default: /tekton/home
       description: Absolute path to the user's home directory.
+    - name: task-image
```
Comment (piyush-garg) [inline task/orka-full/0.1/orka-full.yaml]: This should be removed

File: task/orka-init/0.1/orka-init.yaml [hunk 5]
```diff
@@ -71,9 +76,14 @@ spec:
       description: |
         The name of the key in the VM name config map, which stores the name of the generated VM configuration.
       default: vm-name
+    - name: task-image
```
Comment (piyush-garg) [inline task/orka-init/0.1/orka-init.yaml]: This should be removed

**Review by piyush-garg — CHANGES_REQUESTED** [review]

**Review by vinamra28 — APPROVED** [review]

File: task/orka-init/0.1/orka-init.yaml (hunk 5 above)
Comment (hokadiri) [reply task/orka-init/0.1/orka-init.yaml]: done

File: task/orka-full/0.1/orka-full.yaml (hunk 4 above)
Comment (hokadiri) [reply task/orka-full/0.1/orka-full.yaml]: done

File: task/orka-deploy/0.1/orka-deploy.yaml (hunk 3 above)
Comment (hokadiri) [reply task/orka-deploy/0.1/orka-deploy.yaml]: done

File: task/orka-init/0.1/orka-init.yaml (hunk 2 above)
Comment (hokadiri) [reply task/orka-init/0.1/orka-init.yaml]: done

File: task/orka-full/0.1/orka-full.yaml (hunk 1 above)
Comment (hokadiri) [reply task/orka-full/0.1/orka-full.yaml]: no, remnants of a refactor. removed

**Pull request merged**